apps/site/lib/site_web/controllers/schedule/line/diagram_helpers.ex

```diff
@@ -42,9 +42,29 @@ defmodule SiteWeb.ScheduleController.Line.DiagramHelpers do
   end
 
   def build_stop_list(branches, direction_id) do
+    # Get route_id to check if this is the HH Ferry
+    # Most routes have a merge on the inbound side, 
+    # But the HH ferry merges on the outbound end
+    stops_on_first_branch =
+      branches
+      |> List.first()
+      |> Map.get(:stops)
+
+    route =
+      if stops_on_first_branch,
+        do:
+          stops_on_first_branch
+          |> List.first()
+          |> Map.get(:route),
+        else: nil
+
+    route_id = if route, do: Map.get(route, :id), else: nil
+
+    verified_direction = RouteStop.verify_direction(route_id, direction_id)
+
```

Comment:
It looks like in both instances where we call this function, we have access to the route. So I wonder if that logic could/should be done _before_ calling `build_stop_list`.

In particular, I'm curious if calling `verified_direction = RouteStop.verify_direction(route_id, direction_id)` beforehand, such that you can pass in the resulting `verified_direction` as the second argument here, obviates the need to do these lines of logic here.